### PR TITLE
Add products#show to API v2

### PR DESCRIPTION
**Pull request by kushniryb**

### CODE REVIEW

File: core/app/models/spree/product.rb [hunk 1]
```diff
@@ -125,6 +125,21 @@ class Product < Spree::Base
 
     alias master_images images
 
+    # Can't use any? due to https://github.com/Netflix/fast_jsonapi/issues/259
+    def purchasable?
```
Comment (bbonislawski) [inline core/app/models/spree/product.rb]: Please write specs for new model methods

**Review by bbonislawski — CHANGES_REQUESTED** [review]

File: api/app/controllers/spree/api/v2/storefront/products_controller.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            {
+              resource_serializer: Spree::V2::Storefront::ProductSerializer
+            }
+          end
+
+          def scope
+            Spree::Product.includes(scope_includes)
+          end
+
+          def resource_includes
```
[the remark this answers is not in the document]
Comment (kushniryb) [reply api/app/controllers/spree/api/v2/storefront/products_controller.rb]: @damianlegawiec 
Don't you think it's a little bit too verbose?
Although, we could do something like this
```ruby
def resource_includes
  request_includes || default_resource_includes
end

def default_resource_includes
  {
    ....
  }
end
```
What do you think?

[the remark this answers is not in the document]
Comment (damianlegawiec) [reply api/app/controllers/spree/api/v2/storefront/products_controller.rb]: @kushniryb yep, that's what I've meant :)

[the remark this answers is not in the document]
Comment (kushniryb) [reply api/app/controllers/spree/api/v2/storefront/products_controller.rb]: @damianlegawiec Done

**Review by damianlegawiec — APPROVED** [review]
👍

**Pull request merged**